Assignments/ip2/ip2.md

```diff
@@ -0,0 +1,382 @@
+   */
+  public get id(): string 
+
+  /**
+   * The title of the poster assigned to this area, or undefined if there is not one.
+   */
+  public get title(): string | undefined 
+
+  /**
+   * The poster title: changing this value will emit a 'posterTitleChange' events to listeners
+   * 
+   */
+  public set title(title: string | undefined)
+
+  /**
+   * The image of the poster assigned to this area, or undefined if there is not one.
+   */
+  public get imageContents(): string | undefined 
+
+
+  /**
+   * The image contents of the poster (string representing the contents of the poster file chosen).
+   * Changing this value will emit a 'posterImageContentsChange' event to listeners, and empty the list
+   * of players who starred the poster.
+   */
+  public set imageContents(imageContents: string | undefined) 
+
+  /**
+   * The number of stars of the poster assigned to this area.
+   */
+  public get stars(): number
+
+  /**
+   * The number of stars of the poster assigned to this area.
+   *
+   * Changing this value will emit a ‘posterStarChange' event to listeners
+   */
+  public set stars(stars: number)
+
+  /**
+   * The list of IDs of the players who starred the poster (with its current image)
+   */ 
+  public get playersWhoStarred(): string[]
+
+  /**
+   * Add the specified player ID to the list of players who starred this poster. 
+   */
+  public addPlayerWhoStarred(playerID: string)
+
+  /**
+   * @returns PosterSessionAreaModel that represents the current state of this PosterSessionAreaController
+   */
+  public posterSessionAreaModel(): PosterSessionAreaModel
+{% endhighlight %}
+
+TownController 
+
+{% highlight typescript %}
+    /**
+     * When an interactable's state changes, push that update into the relevant controller, which is assumed
+     * to be either a Viewing Area, a Poster Session Area, or a Conversation Area, and which is assumed to already
+     * be represented by a ViewingAreaController, PosterSessionAreaController or ConversationAreaController that this TownController has.
+     *
+     * If a conversation area transitions from empty to occupied (or occupied to empty), this handler will emit
+     * a conversationAreasChagned event to listeners of this TownController.
+     *
+     * If the update changes properties of the interactable, the interactable is also expected to emit its own
+     * events (@see ViewingAreaController and @see ConversationAreaController and @see PosterSessionAreaController)
+     */
+    this._socket.on('interactableUpdate', interactable => {})
+
+
+  /**
+   * Get the image contents for a specified poster session area (specified via poster session area controller)
+   * @param posterSessionArea the poster session area controller
+   * @returns a promise wrapping the contents of the poster session area's image (i.e. the string)
+   */
+  public async getPosterSessionAreaImageContents(
+    posterSessionArea: PosterSessionAreaController,
+  ): Promise<string> 
```

Comment:
Do they need to implement this or is it just here for docs/specifications? If they need to implement this, we need to update the handout file I believe.